core/src/main/scala/dagr/core/tasksystem/Schedulable.scala

```diff
@@ -48,6 +48,65 @@ trait Schedulable {
+
+  /** A crude attempt at estimating the minimum cores needed by assuming a fixed maximum memory. */
+  private[tasksystem] def minCores(start: Cores = Cores(1), end: Cores = Cores(32), step: Cores = Cores(1), maxMemory: Memory = Memory("64g")): Option[Cores] = {
+    // first check that the search will succeed, then test a range of values for cores
+    this.pickResources(availableResources = new ResourceSet(end, maxMemory)).flatMap { _ =>
+      (start.value to end.value by step.value).flatMap { cores =>
+        val availableResources = new ResourceSet(Cores(cores), maxMemory)
+        this.pickResources(availableResources = availableResources).map(_.cores)
+      }.headOption
+    }
+  }
+
+  /** A crude attempt at estimating the minimum memory needed by assuming a fixed maximum number of cores. */
+  private[tasksystem] def minMemory(start: Memory = Memory("256m"), end: Memory = Memory("64g"), step: Memory = Memory("256m"), maxCores: Cores = Cores(32)): Option[Memory] = {
+    // first check that the search will succeed, then test a range of values for memory
+    this.pickResources(availableResources = new ResourceSet(maxCores, end)).flatMap { _ =>
+      (start.value to end.value by step.value).flatMap { memory =>
+        val availableResources = new ResourceSet(maxCores, Memory(memory))
+        this.pickResources(availableResources = availableResources).map(_.memory)
+      }.headOption
+    }
+  }
+
+  /** A crude attempt at estimating the minimum resources by assuming a fixed ratio of memory per core. */
+  private[tasksystem] def minCoresAndMemory(start: Cores = Cores(1), end: Cores = Cores(32), step: Cores = Cores(1), memoryPerCore: Memory = Memory("1g")): Option[ResourceSet] = {
+    (start.value to end.value by step.value).flatMap { cores =>
+      val availableResources = new ResourceSet(Cores(cores), Memory((memoryPerCore.value * cores).toLong))
+      this.pickResources(availableResources = availableResources)
+    }.headOption
+  }
+
+  /**
+    * Given a non-null ResourceSet representing the maximum available resources return either a
+    * ResourceSet that is a subset of the available resources in which the task can run, or None
+    * if the task cannot run in the maximum resources.
+    *
+    * First estimates the # of cores by assuming an infinite amount of memory, and then estimates
+    * the amount of memory assuming an infinite # of cores.  If either does not yield a value or if
+    * the task cannot run with the combination of the two, then try a various combinations of cores
+    * and memory where we assume a fixed ratio between cores and memory (ex. 1g per core).
+    *
+    * @param maximumResources The maximum system resources available to the task.
+    * @return Either a ResourceSet of the subset of resources that this task can run with, or None
+    */
+  private[core] def minResources(maximumResources: ResourceSet = new ResourceSet(cores=Cores(32), memory=Memory("64g"))): Option[ResourceSet] = {
```

Comment:
The defaults here feel very "no-one will ever need more than 640kb of memory"-esque.  I would either set these way higher (like 1PB and 1000 cores) or set the as some very large multiple of the current system resources.